tensorhive/models/Reservation.py

```diff
@@ -91,19 +72,18 @@ def ends_at(self):
 
     @ends_at.setter  # type: ignore
     def ends_at(self, value):
-        if isinstance(value, str):
-            try:
-                self._ends_at = Reservation.parsed_input_datetime(value)
-            except ValueError:
-                # Catch, but don't propagate at this moment,
-                # let the dev to change it to correct value later
-                self._ends_at = None
-        elif isinstance(value, datetime.datetime):
-            self._ends_at = value
-        else:
-            self._ends_at = None
+        self._ends_at = DateUtils.try_parse_string(value)
+        if not self._ends_at:
```

Comment:
Do we expect any falsy value other than `None` here? If not we could replace this with the following code, so that it explicitly shows the intent of this check.

```suggestion
        if self._ends_at is not None:
```

Reply:
Fixed